internal/extsvc/github/pulls.go

```diff
@@ -245,6 +245,16 @@ type PullRequestReviewThread struct {
 }
 
 type PullRequestCommit struct {
+	Commit Commit
+}
+
+func (p PullRequestCommit) Key() string {
+	return p.Commit.OID
+}
+
+// CommitWithChecks represents check / build status of a commit. When we load the PR
+// from GitHub we fetch the most recent commit into this type to check build status
```

Comment:
nit:

```suggestion
// CommitWithChecks represents check/build status of a commit. When we load the PR
// from GitHub we fetch the most recent commit into this type to check build status.
```

According to [handbook](https://about.sourcegraph.com/handbook/communication/style_guide):
>No spaces between two terms separated by a slash (“a/b”, not “a / b”).

And for Godoc, every docstring should be a complete sentence ends with a period.